homeassistant/components/esphome/manager.py

```diff
@@ -329,6 +329,15 @@ def async_on_state_subscription(
             entity_id, attribute, hass.states.get(entity_id)
         )
 
+    @callback
+    def async_on_state_request(
+        self, entity_id: str, attribute: str | None = None
+    ) -> None:
+        """Forward state for requested entity."""
+        self._send_home_assistant_state(
+            entity_id, attribute, self.hass.states.get(entity_id)
+        )
```

Comment:
Duplicate code in the function above....not sure its enough to consolidate though

Reply:
Yeah I wasnt sure if I could make it less even, unless the above one just calls this new function?
```suggestion
        """Forward state for requested entity."""
        self._send_home_assistant_state(
            entity_id, attribute, self.hass.states.get(entity_id)
        )
```

Reply:
The naming isn't great, but I guess calling the other function would work.

Probably not worth worrying about it since its a single line anyways